### PR TITLE
[exporter/kafka] Allow configuring the acknowledgement behaviour

### PR DESCRIPTION
**Pull request by secustor**
**Description:** <Describe what has changed.>

Allow to configure acknowledgement behavior of the Sarama producer inside of the Kafka exporter.  

**Link to tracking Issue:** #6222

**Testing:** <Describe what testing was performed and which tests were added.>
Adapted unit testing to include the new parameter

**Documentation:** Added field descriptions to the readme

### CODE REVIEW

File: exporter/kafkaexporter/config.go [hunk 1]
```diff
@@ -67,6 +68,13 @@ type Metadata struct {
 type Producer struct {
 	// Maximum message bytes the producer will accept to produce.
 	MaxMessageBytes int `mapstructure:"max_message_bytes"`
+
+	// RequiredAcks Number of Acknowledgements required to assume that a message has been sent.
```
Comment (jcchavezs) [inline exporter/kafkaexporter/config.go:72]:
```suggestion
	// RequiredAcks Number of acknowledgements required to assume that a message has been sent.
```

Comment (jcchavezs) [reply exporter/kafkaexporter/config.go:72]: I would include this link here aswell https://pkg.go.dev/github.com/Shopify/sarama@v1.30.0#RequiredAcks

File: exporter/kafkaexporter/testdata/config.yaml [hunk 2]
```diff
@@ -10,6 +10,7 @@ exporters:
         max: 15
     producer:
       max_message_bytes: 10000000
+      required_acks: -1
```
Comment (jcchavezs) [inline exporter/kafkaexporter/testdata/config.yaml:13]:
```suggestion
      required_acks: -1 # WaitForAll
```

File: exporter/kafkaexporter/config.go [hunk 3]
```diff
@@ -67,6 +68,14 @@ type Metadata struct {
...
 	// Maximum message bytes the producer will accept to produce.
 	MaxMessageBytes int `mapstructure:"max_message_bytes"`
+
+	// RequiredAcks Number of acknowledgements required to assume that a message has been sent.
+	// https://pkg.go.dev/github.com/Shopify/sarama@v1.30.0#RequiredAcks
+	// The options are:
+	//   0 -> NoResponse.  doesn't send any response
+	//   1 -> WaitForLocal. waits for only the local commit to succeed before responding ( default )
+	//   -1 -> WaitForAll. waits for all in-sync replicas to commit before responding.
+	RequiredAcks sarama.RequiredAcks `mapstructure:"required_acks"`
```
Comment (Aneurysm9) [inline exporter/kafkaexporter/config.go:78]: Please add validation that this value is in the enumeration to the `Validate()` method below.

Comment (secustor) [reply exporter/kafkaexporter/config.go:79]: I have added validation to check that `RequiredAcks` is correctly formatted

**Comment by secustor** [comment]
@anuraaga Will you be able to take a look at this PR? Thanks!

**Review by anuraaga — APPROVED** [review]
Sorry for missing this

**Comment by secustor** [comment]
@owais  Could you start the pipeline and review/merge this? Thanks!

**Comment by secustor** [comment]
@jpkrohling  Sry, if you are not the right person to ping, but could you reopen this PR?      

I think this is a worthwhile contribution. Thanks!

**Comment by jpkrohling** [comment]
Reopened.

File: exporter/kafkaexporter/kafka_exporter.go [hunk 4]
```diff
@@ -128,8 +128,7 @@ func newSaramaProducer(config Config) (sarama.SyncProducer, error) {
 	// These setting are required by the sarama.SyncProducer implementation.
 	c.Producer.Return.Successes = true
 	c.Producer.Return.Errors = true
-	// Wait only the local commit to succeed before responding.
-	c.Producer.RequiredAcks = sarama.WaitForLocal
+	c.Producer.RequiredAcks = config.Producer.RequiredAcks
```
Comment (jpkrohling) [inline exporter/kafkaexporter/kafka_exporter.go:131]: Is this the default behavior of the underlying lib? In any case, you need an entry in the changelog about this change.

Comment (secustor) [reply exporter/kafkaexporter/kafka_exporter.go:131]: Yes this is the default. https://github.com/Shopify/sarama/blob/main/config.go#L167

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**